src/support/Maybe.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Maybe monad inspired by https://codewithstyle.info/advanced-functional-programming-in-typescript-maybe-monad/
+ */
+export class Maybe<T> {
```

Comment:
@tpluscode Have you considered being compatible with https://github.com/cbowdon/TsMonad/blob/master/src/maybe.ts?